packages/orn-locator/src/resolve.ts

```diff
@@ -17,15 +18,28 @@ export const locateAll = async(orn: string[], {concurrency = 2}: {concurrency?:
   return await asyncPool(concurrency, orn, locate);
 };
 
-export const search = async(query: string, limit: number = 5, filters: {[key: string]: string | string[]} = {}) => {
-  type Patterns = typeof patterns;
+type Patterns = typeof patterns;
+
+export const search = async(searchClient: SearchClient, query: string, limit: number = 5, type?: string) => {
   const result: {[K in keyof Patterns]?: {name: string; items: Awaited<ReturnType<NonNullable<Patterns[K]['search']>>>}} = {};
 
-  for (const [key, pattern] of Object.entries(patterns)) {
-    const innerResult = await pattern.search?.(query, limit, filters);
+  const searchPattern = async(key: keyof Patterns, pattern: Patterns[keyof Patterns]) => {
+    const innerResult = await pattern.search?.(searchClient, query, limit);
     if (innerResult && innerResult.length > 0) {
       result[key as keyof Patterns] = {items: innerResult as any, name: pattern.name};
     }
+  };
+
+  if (type) {
+    if (type in patterns) {
```

Comment:
shouldn't it not be looping if you pass a type? this seems like its doing the same thing as the condition below

Reply:
Really it only needs the 1 type but I had this at some point and I can revert if that's what you prefer:

```
  for (const [key, pattern] of Object.entries(patterns)) {
    if ((type && key !== type) || (!type && pattern.excludeFromDefaultSearch)) { continue; }

    const innerResult = await pattern.search?.(searchClient, query, limit);
    if (innerResult && innerResult.length > 0) {
      result[key as keyof Patterns] = {items: innerResult as any, name: pattern.name};
    }
  }
```

Reply:
I'm confused. Doesn't the existing code just overwrite the type variable with a loop over all patterns?

Reply:
No it's an if statement not a for statement ignore me 🤦‍♂️